File: season4/week3/bada/solution_3.py
def solution(k, tangerine):
    answer = 0
    a={}
    for i in tangerine:
        if i in a:
            a[i]+=1
        else:
            a[i]=1
    a = dict(sorted(a.items(), key=lambda x: x[1], reverse=True))
    for i in a:
        if k<=0:
            return answer
        k-=a[i]
        answer+=1
    return answer

File: season4/week3/sumin/week3-sumin/src/Solution3.java
import java.util.*;

public class Solution3 {
    public int solution(int k, int[] tangerine) {
        int answer = 0;
        Arrays.sort(tangerine);

        Map<Integer, Integer> map = new HashMap<>();
        for (int i = 0; i < tangerine.length; i++) {
            map.put(tangerine[i], map.getOrDefault(tangerine[i], 0) + 1);
        }
        List<Integer> keySet = new ArrayList<>(map.keySet());
        keySet.sort((o1, o2) -> map.get(o2).compareTo(map.get(o1)));

        for (int i : keySet) {
            k -= map.get(i);
            answer++;
            if (k <= 0) return answer;
        }

        return answer;
    }
}
